### PR TITLE
Nats messaging refactor

### PR DESCRIPTION
**Pull request by rhlsthrm**
## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

**Comment by ArjunBhuptani** [comment]
Still a LOT of cleanup to do -- will finish this tomorrow

**Comment by kthomas** [comment]
> Still a LOT of cleanup to do -- will finish this tomorrow

Following along, looking good...

File: modules/messaging/src/service.ts [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
     private readonly config: MessagingConfig,
     private readonly messagingServiceKey: string,
-    private readonly bearerToken?: string,
+    private readonly clientAuthCallback: () => Promise<string>,
   ) {
-    this.log = new Logger(`MessagingService`, config.logLevel);
-    this.log.debug(`Created with config: ${JSON.stringify(config, null, 2)}`);
+    this.log = config.logger || nullLogger;
+    this.log.debug(`Created NatsMessagingService with config: ${JSON.stringify(config, null, 2)}`);
+    this.bearerToken = config.bearerToken;
```
Comment (kthomas) [inline modules/messaging/src/service.ts]: `config.bearerToken`? This smells a little...

**Comment by kthomas** [comment]
This PR is epic 🙌

**Pull request merged**